File: src/main/java/com/dochiri/hexagonal/domain/user/vo/UserId.java
package com.dochiri.hexagonal.domain.user.vo;

import java.util.UUID;

import static java.util.Objects.*;

public record UserId(
        String value
) {
    public UserId {
        requireNonNull(value, "UserId는 null일 수 없습니다.");
        if (value.isBlank()) {
            throw new IllegalArgumentException("UserId는 비어있을 수 없습니다.");
        }
    }

    public static UserId from(String value) {
        return new UserId(value);
    }

    public static UserId newId() {
        return new UserId(UUID.randomUUID().toString());
    }

}

File: src/main/kotlin/com/dochiri/hexagonal/domain/user/vo/UserId.kt
package com.dochiri.hexagonal.domain.user.vo

import java.util.UUID

@JvmInline
value class UserId private constructor(val value: String) {
    companion object {
        fun generate(): UserId = UserId(UUID.randomUUID().toString())

        fun from(raw: String): UserId {
            val normalized = raw.trim()
            require(normalized.isNotBlank()) { "UserId는 비어 있을 수 없습니다." }
            return UserId(normalized)
        }

        fun newId(): UserId = generate()
    }
}